quarkchain/cluster/shard.py

```diff
@@ -560,15 +559,30 @@ def broadcast_tx_list(self, tx_list, source_peer=None):
         if self.state.db.contain_minor_block_by_hash(block.header.get_hash()):
             return
 
-        if not self.state.db.contain_minor_block_by_hash(
-            block.header.hash_prev_minor_block
+        prev_hash, prev_header = block.header.hash_prev_minor_block, None
+        if prev_hash in self.state.new_block_pool:
+            prev_header = self.state.new_block_pool[prev_hash].header
+        else:
+            prev_header = self.state.db.get_minor_block_header_by_hash(prev_hash)
+        if prev_header is None:  # Missing prev
+            return
+
+        # Sanity check on timestamp and block height
+        if (
+            block.header.create_time
+            > time_ms() // 1000 + ALLOWED_FUTURE_BLOCKS_TIME_BROADCAST
         ):
-            if block.header.hash_prev_minor_block not in self.state.new_block_pool:
-                return
+            return
+        if (
+            self.state.header_tip
+            and self.state.header_tip.height - block.header.height
+            > self.state.shard_config.max_stale_minor_block_height_diff
+        ):
+            return
```

Comment:
it's essentially the same check when adding the block

https://github.com/QuarkChain/pyquarkchain/blob/178d8976a26989e13bd1aee0c9dfb6221cb2260a/quarkchain/cluster/shard_state.py#L812-L841

just add it here as a prerequisite